main.go

```diff
@@ -174,305 +109,170 @@ func parseConfig(args []string) (*config, error) {
 
 	// this fails if there are flags re-defined with the same name.
 	if err := flag.CommandLine.Parse(args); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if flag.NFlag() == 0 {
 		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
 		flag.PrintDefaults()
-		return nil, flag.ErrHelp
-	}
-
-	cfg := &config{
-		file:                 *flagFile,
-		line:                 *flagLine,
-		structName:           *flagStruct,
-		fieldName:            *flagField,
-		offset:               *flagOffset,
-		all:                  *flagAll,
-		output:               *flagOutput,
-		write:                *flagWrite,
-		quiet:                *flagQuiet,
-		clear:                *flagClearTags,
-		clearOption:          *flagClearOptions,
-		transform:            *flagTransform,
-		sort:                 *flagSort,
-		valueFormat:          *flagFormatting,
-		override:             *flagOverride,
-		skipUnexportedFields: *flagSkipUnexportedFields,
+		return nil, nil, flag.ErrHelp
+	}
+
+	p := &pass{
+		file:       *flagFile,
+		line:       *flagLine,
+		structName: *flagStruct,
+		fieldName:  *flagField,
+		offset:     *flagOffset,
+		all:        *flagAll,
+		output:     *flagOutput,
+		write:      *flagWrite,
+		quiet:      *flagQuiet,
+	}
+
+	m := &modifytags.Mutation{
+		Clear:                *flagClearTags,
+		ClearOption:          *flagClearOptions,
+		Transform:            *flagTransform,
+		Sort:                 *flagSort,
+		ValueFormat:          *flagFormatting,
+		Override:             *flagOverride,
+		SkipUnexportedFields: *flagSkipUnexportedFields,
 	}
 
 	if *flagModified {
-		cfg.modified = os.Stdin
+		p.modified = os.Stdin
 	}
 
 	if *flagAddTags != "" {
-		cfg.add = strings.Split(*flagAddTags, ",")
+		m.Add = strings.Split(*flagAddTags, ",")
 	}
 
 	if *flagAddOptions != "" {
-		cfg.addOptions = strings.Split(*flagAddOptions, ",")
+		m.AddOptions = make(map[string][]string)
```

Comment:
Factor this out into a helper, since it really shares the same concern with RemoveOptions below. "parseOptions", perhaps?